syntax = "proto3";

package routing;

// The routing service definition
service RoutingService {
  // Health check
  rpc HealthCheck (HealthRequest) returns (HealthResponse) {}
  
  // Find routes between two locations
  rpc FindRoute (RouteRequest) returns (RouteResponse) {}
}

// Health check request (empty)
message HealthRequest {}

// Health check response
message HealthResponse {
  string status = 1;
  string message = 2;
}

// Request for finding a route
message RouteRequest {
  double start_lon = 1;
  double start_lat = 2;
  double end_lon = 3;
  double end_lat = 4;
  int32 max_transfers = 5;
  double walking_cutoff = 6;
  repeated string restricted_modes = 7;
  RoutingWeights weights = 8;
  int32 top_k = 9;
}

// Routing weights for journey ranking
message RoutingWeights {
  double time = 1;
  double cost = 2;
  double walk = 3;
  double transfer = 4;
}

// Response containing all found journeys as JSON
message RouteResponse {
  string journeys_json = 1;  // Complete enriched journey data as JSON
}
